include/net/socket_can.h

```diff
@@ -47,6 +48,20 @@ struct sockaddr_can {
 	int         can_ifindex;
 };
 
+/**
+ * CAN L2 driver API. Used by socket CAN.
+ */
+struct canbus_api {
```

Comment:
This probably should be called canbus_driver_api.

Reply:
This is not the driver. This is just a wrapper.

Reply:
> This is not the driver. This is just a wrapper.

Then it's confusing and likely unneeded wrapper. A lot in this PR seems to be done to twist net/ip/ to push unrelated protocol thru it.